### PR TITLE
[Doc]Update OAP docs for 0.8.2 release

### PR DESCRIPTION
**Pull request by HongW2019**

### CODE REVIEW

File: oap-cache/oap/docs/User-Guide.md [hunk 1]
```diff
@@ -17,22 +17,22 @@ SQL Index and Data Source Cache on Spark requires a working Hadoop cluster with
...
 
 ### Spark Configurations
 
 Users usually test and run Spark SQL or Scala scripts in Spark Shell,  which launches Spark applications on YRAN with ***client*** mode. In this section, we will start with Spark Shell then introduce other use scenarios. 
 
-Before you run ` . $SPARK_HOME/bin/spark-shell `, you need to configure Spark for integration. You need to add or update the following configurations in the Spark configuration file `$SPARK_HOME/conf/spark-defaults.conf` on your working node.
+Before you run ` . $SPARK_HOME/bin/spark-shell `, make sure `oap-common-0.8.2-with-spark-2.4.4.jar` has already been put into the directory `$SPARK_HOME/jars/`, then you need to configure Spark for integration. You need to add or update the following configurations in the Spark configuration file `$SPARK_HOME/conf/spark-defaults.conf` on your working node.
```
Comment (zhixingheyi-tian) [inline oap-cache/oap/docs/User-Guide.md:26]: move "oap-common-0.8.2-with-spark-2.4.4.jar" to the below config item

Comment (HongW2019) [reply oap-cache/oap/docs/User-Guide.md]: Modified.

File: oap-shuffle/RPMem-shuffle/README.md [hunk 2]
```diff
@@ -430,7 +430,7 @@ mkdir build
 cd build
 cmake -DCMAKE_INSTALL_PREFIX=/usr/local -DBUILD_EXAMPLES=1 -DBUILD_TESTS=1 ..
 make all && make install
-git clone -b v0.8.1-spark-2.4.4 https://github.com/intel-bigdata/OAP.git
+git clone -b v0.8.2-spark-2.4.4 https://github.com/intel-bigdata/OAP.git
 cd OAP/oap-shuffle/RPMem-shuffle
```
Comment (zhixingheyi-tian) [inline oap-shuffle/RPMem-shuffle/README.md:434]: {version}

Comment (HongW2019) [reply oap-shuffle/RPMem-shuffle/README.md:434]: modified

File: .gitignore [hunk 3]
```diff
@@ -0,0 +1,6 @@
+**/target/**
+*.class
+*.iml
+*.sh
+# IDEA config files
+.idea/
```
Comment (zhixingheyi-tian) [inline .gitignore:6]: need to add these filters？

File: oap-cache/oap/docs/User-Guide.md [hunk 4]
```diff
@@ -569,27 +573,40 @@ spark.sql.oap.orc.data.cache.enable             false
...
+
 
 ## Run TPC-DS Benchmark
 
 This section provides instructions and tools for running TPC-DS queries to evaluate the cache performance of various configurations. The TPC-DS suite has many queries and we select 9 I/O intensive queries to simplify performance evaluation.
 
-We created some tool scripts [OAP-TPCDS-TOOL.zip](https://github.com/Intel-bigdata/OAP/releases/download/v0.8.1-spark-2.4.4/OAP-TPCDS-TOOL.zip) to simplify running the workload. If you are already familiar with TPC-DS data generation and running a TPC-DS tool suite, skip our tool and use the TPC-DS tool suite directly.
+We created some tool scripts [OAP-TPCDS-TOOL.zip](https://github.com/Intel-bigdata/OAP/releases/download/v0.8.2-spark-2.4.4/OAP-TPCDS-TOOL.zip) to simplify running the workload. If you are already familiar with TPC-DS data generation and running a TPC-DS tool suite, skip our tool and use the TPC-DS tool suite directly.
```
Comment (zhixingheyi-tian) [inline oap-cache/oap/docs/User-Guide.md:601]: Are OAP-TPCDS-TOOL.zip internals changed ?

File: .gitignore (hunk 3 above)
Comment (HongW2019) [reply .gitignore:6]: Yes

File: oap-cache/oap/docs/User-Guide.md (hunk 4 above)
Comment (HongW2019) [reply oap-cache/oap/docs/User-Guide.md:601]: Yes, we will upload the updated it to release page.

File: oap-cache/oap/docs/User-Guide.md [hunk 5]
```diff
@@ -300,8 +300,8 @@ Change the values of `spark.executor.instances`, `spark.sql.oap.fiberCache.persi
 
 #### Choose additional configuration options
 
-Optimize your environment by choosing a DCPMM caching strategy (guava, non-evictable, vmemcache, external cache using plasma). Following table shows features of each cache.
-| guava | non-evictable | vmemcache | external cache |
+Optimize your environment by choosing a DCPMM caching strategy (guava, noevict, vmemcache, external cache using plasma). Following table shows features of each cache.
+| guava | noevict | vmemcache | external cache |
```
Comment (zhixingheyi-tian) [inline oap-cache/oap/docs/User-Guide.md:304]: cc @yma11  , Is  it correct for this change

File: oap-cache/oap/docs/User-Guide.md [hunk 6]
```diff
@@ -487,67 +489,69 @@ spark.sql.oap.fiberCache.memory.manager                 offheap
...
 # enable numa
 spark.yarn.numa.enabled                                  true
-spark.executorEnv.MEMKIND_ARENA_NUM_PER_KIND             1
 spark.memory.offHeap.enabled                             false
 # DCPMM capacity per executor
 spark.sql.oap.fiberCache.persistent.memory.initial.size  256g
-# Reserved space per executor
-spark.sql.oap.fiberCache.persistent.memory.reserved.size 50g
+# according to your cluster
+spark.sql.oap.cache.guardian.memory.size                 10g
```
Comment (zhixingheyi-tian) [inline oap-cache/oap/docs/User-Guide.md:518]: Is it for memkind or Vmemcache?

**Comment by zhixingheyi-tian** [comment]
@HongW2019  
As to the modification of key configurations, make sure these has been verified.

File: oap-cache/oap/docs/User-Guide.md (hunk 6 above)
Comment (jikunshang) [reply oap-cache/oap/docs/User-Guide.md:518]: memkind don't need this configuration.

File: oap-cache/oap/docs/User-Guide.md (hunk 5 above)
Comment (jikunshang) [reply oap-cache/oap/docs/User-Guide.md:304]: it's correct.

File: oap-cache/oap/docs/User-Guide.md [hunk 7]
```diff
@@ -487,67 +489,71 @@ spark.sql.oap.fiberCache.memory.manager                 offheap
...
 The rest configurations can refer to the configurations of  [Use DRAM Cache](#use-dram-cache) 
 
-2. DCPMM(`pm`) as cache media, `guava` strategy as index, and data cache back end. 
+2. DCPMM as cache media, `vmem` strategy as index & data cache backend. 
 
 ```
 spark.sql.oap.index.data.cache.separation.enable        true
 spark.oap.cache.strategy                                mix
-spark.sql.oap.fiberCache.memory.manager                 pm
+spark.sql.oap.fiberCache.memory.manager                 tmp
```
Comment (HongW2019) [inline oap-cache/oap/docs/User-Guide.md:497]: @yma11 @yeyuqiang Can we here use " DCPMM as cache media, `vmem` strategy as index & data cache backend. "? if so , why queries failed when not adding "spark.sql.oap.fiberCache.memory.manager  tmp"?

**Pull request merged**